lib/court_data_adaptor/errors.rb

```diff
@@ -31,6 +34,26 @@ def retrieve_user_facing_string
       end
     end
 
+    class ErrorCodeParser
+      def self.call(cda_response)
+        return unless cda_response
+
+        body = cda_response.body
+        error_response = body.is_a?(String) ? JSON.parse(body) : body
+        return unless error_response['error_codes']
+
+        error_response['error_codes'].filter_map { |code| build_message(code) }.join(" ").presence
+      rescue StandardError
```

Comment:
is this capturing the `JSON::ParserError`, 
in that case,  maybe we could check `json_string?` , something like this

```
def json_string?(str)
  JSON.parse(str)
  true
rescue JSON::ParserError
  false
end
```